src/pages/test.js

```diff
@@ -0,0 +1,33 @@
+import useSWR from 'swr'
+import {supabase} from '../utils/supabaseClient'
+
+// @todo somehow pass this query in through useSWR to the fetcher.
+const QUERY = supabase.from('channels').select('*')
+
+// We overwrite the default fetcher because supabase returns {data, error},
+// which we need to handle.
+// SWR expects either some data or to throw an error.
+async function supabaseFetcher() {
+	const {data, error} = await QUERY
+	if (error) throw new Error(error)
+	return data
+}
+
+export default function Test() {
+	const {data, error} = useSWR('a unique identifier for caching purposes', supabaseFetcher)
```

Comment:
we want to pass queries (like `QUERY`) directly through `useSWR` but I dunno how. Tried this:

```js
useSWR(['unique key', myQueryFunction]], fetcher)
...
function fetcher(...args) {
// args[1] will be the query fn
```

but then it fetches in an infinite loop.